libs/damap/src/lib/widgets/internal-storage-table/internal-storage-table.component.ts

```diff
@@ -0,0 +1,144 @@
+import { MatDialog } from '@angular/material/dialog';
+import { InternalStorageDialogComponent } from '../../components/admin/internal-storage-dialog/internal-storage-dialog.component';
+import { DeleteStorageWarningDialogComponent } from './dialog/delete-storage-warning-dialog.component';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'damap-internal-storage-table',
+  templateUrl: './internal-storage-table.component.html',
+  styleUrls: ['./internal-storage-table.component.css'],
+})
+export class InternalStorageTableComponent implements AfterViewInit, OnChanges {
+  @Output() selectInternalStorage = new EventEmitter<number>();
+
+  constructor(
+    private backendService: BackendService,
+    private feedbackService: FeedbackService,
+    private dialog: MatDialog,
+    private translateService: TranslateService,
+  ) {}
+
+  @Input() internalStorages: InternalStorage[] = [];
+  dataSource = new MatTableDataSource<InternalStorage>();
+
+  readonly tableHeaders: string[] = [
+    'url',
+    'storageLocation',
+    'backupLocation',
+    'active',
+    'actions',
+  ];
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.internalStorages) {
+      this.dataSource.data = this.internalStorages;
+    }
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+
+    this.dataSource.filterPredicate = (data: InternalStorage, filter: string) =>
+      data.url?.toLowerCase().includes(filter.trim().toLowerCase());
+  }
+
+  toggleActivation(id: number) {
+    const storage = this.internalStorages.find(s => s.id === id);
+    const storageCopy = { ...storage };
+    storageCopy.active = !storage.active;
+    this.backendService.updateInternalStorage(storageCopy).subscribe(() => {
+      this.internalStorages = this.internalStorages.map(s =>
+        s.id === id ? storageCopy : s,
+      );
+      this.dataSource.data = this.internalStorages;
+    });
+  }
+
+  deleteStorage(id: number) {
+    this.dialog
+      .open(DeleteStorageWarningDialogComponent, {
+        data: { deleteType: 'storage' },
+      })
+      .afterClosed()
+      .subscribe({
+        next: response => {
+          if (response) {
+            this.backendService.deleteInternalStorage(id).subscribe(
+              () => {
+                this.internalStorages = this.internalStorages.filter(
+                  s => s.id !== id,
+                );
+                this.dataSource.data = this.internalStorages;
+                this.editTranslations(null);
+              },
+              error => {
+                if (error.status === 409) {
+                  this.feedbackService.error(
+                    this.translateService.instant(
+                      'http.error.storageErrors.stillInUse',
+                    ),
+                  );
```

Comment:
comment/shelve: This works for the current use case. In general, this will have to be adapted to show all messages from the back end. Might as well have them translated already in the back end, when user management is in place (so we can read the preferred language of the user) or when support for `accept-language` header is implemented.